optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/PlannerBenchmarkRunner.java

```diff
@@ -169,13 +172,41 @@ private void warmUp() {
         logger.info("================================================================================");
     }
 
-    private void backupTerminationConfigs(Map<SolverBenchmarkResult, TerminationConfig> originalTerminationConfigMap) {
+    private Map<SolverBenchmarkResult, ConfigBackup> backupBenchmarkConfig(Map<ProblemBenchmarkResult, List<ProblemStatistic>> originalProblemStatisticMap) { // backup & remove stats, backup termination config
+        Map<SolverBenchmarkResult, ConfigBackup> configBackupMap = new HashMap<SolverBenchmarkResult, ConfigBackup>(plannerBenchmarkResult.getSolverBenchmarkResultList().size());
         for (SolverBenchmarkResult solverBenchmarkResult : plannerBenchmarkResult.getSolverBenchmarkResultList()) {
             TerminationConfig originalTerminationConfig = solverBenchmarkResult.getSolverConfig().getTerminationConfig();
-            if (!originalTerminationConfigMap.containsKey(solverBenchmarkResult)) {
-                originalTerminationConfigMap.put(solverBenchmarkResult, originalTerminationConfig);
+            ConfigBackup configBackup = new ConfigBackup(originalTerminationConfig);
+            for (SingleBenchmarkResult singleBenchmarkResult : solverBenchmarkResult.getSingleBenchmarkResultList()) {
+                List<PureSingleStatistic> originalPureSingleStatisticList = singleBenchmarkResult.getPureSingleStatisticList();
+                List<PureSingleStatistic> singleBenchmarkStatisticPutResult = configBackup.getPureSingleStatisticMap().put(singleBenchmarkResult, originalPureSingleStatisticList);
+                if (singleBenchmarkStatisticPutResult != null) {
+                    throw new IllegalStateException("SingleBenchmarkStatisticMap of ConfigBackup ( " + configBackup
+                            + " ) already contained key ( " + singleBenchmarkResult + " ) with value ( "
+                            + singleBenchmarkStatisticPutResult + " ).");
+                }
+                singleBenchmarkResult.setPureSingleStatisticList(Collections.<PureSingleStatistic>emptyList());
+
+                ProblemBenchmarkResult problemBenchmarkResult = singleBenchmarkResult.getProblemBenchmarkResult();
+                List<ProblemStatistic> originalProblemStatisticList = problemBenchmarkResult.getProblemStatisticList();
+                if (!originalProblemStatisticMap.containsKey(problemBenchmarkResult)) { // TODO: After Java 8, do Map#putIfAbsent
+                    List<ProblemStatistic> problemStatisticPutResult = originalProblemStatisticMap.put(problemBenchmarkResult, originalProblemStatisticList);
+                    if (problemStatisticPutResult != null && !originalProblemStatisticList.isEmpty()) {
```

Comment:
It's a put(), not a get(). Whether the actual value is null doesn't matter. See put()'s javadoc return value :)

Furthermore, for a get() that returns null but the key does exist, an isEmpty() check wouldn't suffice as it would be be broken if the map contains other keys too.